cpmpy/solvers/choco.py

```diff
@@ -0,0 +1,582 @@
+                for i in range(len(vars)):
+                    if isinstance(vars[i], int):
+                        vars[i] = self.chc_model.intvar(vars[i], vars[i])  # convert to "variable"
+                    elif not isinstance(vars[i], IntVar):
+                        raise Exception(f"Choco cannot accept alldifferent with: {vars[i]}")
+                return self.chc_model.all_different(vars)
+            elif cpm_expr.name == 'alldifferent_except0':
+                vars = self.solver_vars(cpm_expr.args)
+                for i in range(len(vars)):
+                    if isinstance(vars[i], int):
+                        vars[i] = self.chc_model.intvar(vars[i], vars[i])  # convert to "variable"
+                    elif not isinstance(vars[i], IntVar):
+                        raise Exception(f"Choco cannot accept alldifferent_except0 with: {vars[i]}")
+                return self.chc_model.all_different_except_0(vars)
+            elif cpm_expr.name == 'allequal':
+                vars = self.solver_vars(cpm_expr.args)
+                for i in range(len(vars)):
+                    if isinstance(vars[i], int):
+                        vars[i] = self.chc_model.intvar(vars[i], vars[i])  # convert to "variable"
+                    elif not isinstance(vars[i], IntVar):
+                        raise Exception(f"Choco cannot accept allequal with: {vars[i]}")
+                return self.chc_model.all_equal(vars)
+            elif cpm_expr.name == 'table':
+                assert (len(cpm_expr.args) == 2)  # args = [array, table]
+                array, table = self.solver_vars(cpm_expr.args)
+                return self.chc_model.table(array, table)
+            elif cpm_expr.name == 'InDomain':
+                assert (len(cpm_expr.args) == 2)  # args = [array, table]
+                expr, table = self.solver_vars(cpm_expr.args)
+                return self.chc_model.member(expr, table)
+            elif cpm_expr.name == "cumulative":
+                start, dur, end, demand, cap = self.solver_vars(cpm_expr.args)
+                # Everything given to cumulative in Choco needs to be a variable.
+                # Convert demands to variables
+                if is_num(demand):  # Create list for demand per task
+                    demand = [demand] * len(start)
+                if isinstance(demand, _NumVarImpl):
+                    demand = self.solver_vars(demand)
+                else:
+                    demand = [self.chc_model.intvar(d, d) for d in demand]  # Create variables for demand
+                # Create task variables. Choco can create them only one by one
+                tasks = [self.chc_model.task(s, d, e) for s, d, e in zip(start, dur, end)]
+                # Convert capacity to variable
+                # Choco needs result to be a variable
+                if isinstance(cap, int):
+                    capacity = self.chc_model.intvar(cap, cap)  # convert to "variable"
```

Comment:
Could be usefull to make a helper function for making variables from ints? Such as `def to_var(self, int)`